File: backends/php-laravel/app/Infrastructure/Models/JwtHistory.php
<?php

/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */

declare(strict_types=1);

namespace App\Infrastructure\Models;

use App\Support\Models\AppTransactionModel;
use Database\Factories\JwtHistoryFactory;

/**
 * JWT履歴Modelクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Infrastructure\Models
 */
class JwtHistory extends AppTransactionModel
{
    /**
     * {@inheritdoc}
     */
    protected $table = 'jwt_histories';

    /**
     * {@inheritdoc}
     */
    protected $casts = [
        'issue_at' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
        'deleted_at' => 'datetime',
    ];

    /**
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function client()
    {
        return $this->belongsTo(Client::class);
    }

    /**
     * {@inheritdoc}
     */
    protected static function newFactory()
    {
        return JwtHistoryFactory::new();
    }
}

File: backends/ruby-rails/app/infrastructure/model/jwt_history.rb
# frozen_string_literal: true
#
# JWT 履歴モデルを定義するインフラストラクチャモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Infrastructure
  module Model
    # jwt_histories テーブルに対応する ActiveRecord モデルです。
    class JwtHistory < ActiveRecord::Base
      self.table_name = "jwt_histories"
    end
  end
end
